snappi_ixnetwork/protocolmetrics.py

```diff
@@ -0,0 +1,122 @@
+from snappi_ixnetwork.timer import Timer
+
+
+class ProtocolMetrics(object):
+    # Currently paging of the statistic view is not handled
+    # TODO Need to enhance when device groups statistics reach
+    # more than one page.
+
+    _SUPPORTED_PROTOCOLS_ = [
+        "bgpv4"
+    ]
+    _PROTO_NAME_MAP_ = {
+        "bgpv4": 'BGP Peer Per Port',
+        "bgpv4_drill": 'BGP Peer Drill Down'
+    }
+
+    _RESULT_COLUMNS = {
+        "bgpv4": [
+            ("name", "Device Group", str),
+            ("sessions_total", "Sessions Total", int),
+            ("sessions_up", "Sessions Up", int),
+            ("sessions_down", "Sessions Down", int),
+            ("sessions_not_started", "Sessions Not Started", int),
+            ("routes_advertised", "Routes Advertised", int),
+            ("routes_withdrawn", "Routes Withdrawn", int),
+        ]
+    }
+
+    def __init__(self, ixnetworkapi):
+        self._api = ixnetworkapi
+        self.columns = []
+        self.device_names = []
+
+    def get_supported_protocols(self):
+        """
+        Return the protocols that are supported currently
+        """
+        return self._SUPPORTED_PROTOCOLS_
+
+    def _get_per_port_stat_view(self, protocol):
+        protocol_name = self._PROTO_NAME_MAP_.get(protocol)
+        if protocol_name is None:
+            raise NotImplementedError(
+                "{} is Not Implemented".format(protocol)
+            )
+        try:
+            table = self._api.assistant.StatViewAssistant(protocol_name, 10)
```

Comment:
let's not hardcode the timeout. let's call it metrics_timeout or something and use that variable instead. Also, for now we can use same timeout variable for all kinds of metrics.

Reply:
BTW, don't keep the default timeout to 10. Maybe 30.
We can reduce it later, but for now I'd like to be conservative. What do you think ? (you can add a TODO there too)